federation/pkg/federation-controller/util/meta.go

```diff
@@ -34,8 +34,28 @@ func CopyObjectMeta(obj api_v1.ObjectMeta) api_v1.ObjectMeta {
 	}
 }
 
+// Deep copies cluster-independent, user provided data from the given ObjectMeta struct. If in
+// the future the ObjectMeta structure is expanded then any field that is not populated
+// by the api server should be included here.
+func DeepCopyObjectMeta(obj api_v1.ObjectMeta) api_v1.ObjectMeta {
+	copyMeta := CopyObjectMeta(obj)
+	if obj.Labels != nil {
```

Comment:
Add a comment regarding why is this required?
CopyObjectMata should set the right labels and annotations